### ISSUE TITLE
`createChannel()` can throw with the callback API

### ISSUE THREAD

**Issue by sakai135** [squaremo/amqp.node#129]
In the callback API, the `createChannel()` can throw an error instead of returning the error in the callback.

```javascript
var amqp = require('amqplib/callback_api');
amqp.connect('amqp://localhost?channelMax=1', function(err, conn) {
  conn.createChannel(function(err) {
    if (err) {
      return console.log(err);
    }
    console.log('callback created 1');
    conn.createChannel(function(err) {
      // does not get called
      if (err) {
        return console.log(err);
      }
      console.log('callback created 2');
    }); // this immediately throws
  });
});
```

```
callback created 1

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: No channels left to allocate
    at Connection.C.freshChannel
```

**Comment by squaremo** [comment]
Yes you are quite right. (And similarly with the promise API, if it doesn't reject the promise.)

**Comment by squaremo** [comment]
Fixed in 5770c352859fd04fff2188186c52f30faae02420
